cuda/solver/lower_trs_kernels.cu

```diff
@@ -107,48 +108,50 @@ void generate(std::shared_ptr<const CudaExecutor> exec,
 
         ValueType one = 1.0;
 
-        GKO_ASSERT_NO_CUSPARSE_ERRORS(
-            cusparseSetPointerMode(handle, CUSPARSE_POINTER_MODE_HOST));
-        cusparse::buffer_size_ext(
-            handle, solve_struct->algorithm, CUSPARSE_OPERATION_NON_TRANSPOSE,
-            CUSPARSE_OPERATION_TRANSPOSE, matrix->get_size()[0], num_rhs,
-            matrix->get_num_stored_elements(), &one, solve_struct->factor_descr,
-            matrix->get_const_values(), matrix->get_const_row_ptrs(),
-            matrix->get_const_col_idxs(), nullptr, num_rhs,
-            solve_struct->solve_info, solve_struct->policy,
-            &solve_struct->factor_work_size);
-
-        // allocate workspace
-        if (solve_struct->factor_work_vec != nullptr) {
-            GKO_ASSERT_NO_CUDA_ERRORS(cudaFree(solve_struct->factor_work_vec));
+        {
+            cusparse_pointer_mode_guard pm_guard(handle);
+            device_guard dev_guard(exec->get_device_id());
```

Comment:
is `device_guard` needed here? CudaExecutor run should already have `device_guard`

Reply:
Mike is right, in fact the device_guard is already taken care of by CudaExecutor::run.
https://github.com/ginkgo-project/ginkgo/blob/1b9b919d566cab75cee6291f3adf16cd922f06fb/cuda/base/executor.cpp#L189

That was my bad.

Reply:
It would be needed only for things which are not called through `CudaExecutor::run()`, i.e. non kernels. Like functions you would call in the constructor if you have any.

Reply:
Ah, okay. I have updated this now.